### PR TITLE
[ISSUE #7551] Reuse helper methods from Netty to free direct byte buffer

### LINKED ISSUE #7551: [Enhancement] Git rid of internal sun.nio.ch.DirectBuffer
### Before Creating the Enhancement Request

- [X] I have confirmed that this should be classified as an enhancement rather than a bug/feature.

### Summary

Remove dependency on sun.nio.ch.DirectBuffer

### Motivation

Get rid of dependency on sun.nio.ch.DirectBuffer,  making codebase JDK 9+ compatible

### Describe the Solution You'd Like

Reuse assets from netty

### Describe Alternatives You've Considered

Repeat work offered by netty

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by lizhanhui**
Make codebase JDK 9+ compatible

fix #7551

### CODE REVIEW

File: proxy/src/main/java/org/apache/rocketmq/proxy/service/message/LocalRemotingCommand.java [hunk 1]
```diff
@@ -32,6 +32,7 @@ public static LocalRemotingCommand createRequestCommand(int code, CommandCustomH
         cmd.writeCustomHeader(customHeader);
         cmd.setExtFields(new HashMap<>());
         setCmdVersion(cmd);
+        cmd.makeCustomHeaderToNet();
```
Comment (joeCarf) [inline proxy/src/main/java/org/apache/rocketmq/proxy/service/message/LocalRemotingCommand.java:35]: seems that this is irrelevant to the commit

Comment (joeCarf) [reply proxy/src/main/java/org/apache/rocketmq/proxy/service/message/LocalRemotingCommand.java:35]: I saw your commit in another pr, maybe you can split them

**Review by joeCarf — APPROVED** [review]

**Review by RongtongJin — APPROVED** [review]

**Review by mxsm — APPROVED** [review]
LGTM

File: proxy/src/main/java/org/apache/rocketmq/proxy/service/message/LocalRemotingCommand.java (hunk 1 above)
Comment (lizhanhui) [reply proxy/src/main/java/org/apache/rocketmq/proxy/service/message/LocalRemotingCommand.java:35]: Include it to make the Test CI Pipeline pass

**Review by lizhimins — COMMENTED** [review]
OK

**Pull request merged**